lib/mayaUsd/ufe/wrapUtils.cpp

```diff
@@ -194,6 +217,47 @@ int ufePathToInstanceIndex(const std::string& ufePathString)
 #endif
 }
 
+bool isAttributeEditAllowed(const PXR_NS::UsdAttribute& attr)
+{
+    // get the property spec in the edit target's layer
+    const auto& prim = attr.GetPrim();
+    const auto& stage = prim.GetStage();
+    const auto& editTarget = stage->GetEditTarget();
+
+    // get the index to edit target layer
+    const auto targetLayerIndex = findLayerIndex(attr, editTarget.GetLayer());
+
+    // get the strength-ordered ( strong-to-weak order ) list of property specs that provide
+    // opinions for this property.
+    const auto& propertyStack = attr.GetPropertyStack();
+
+    // HS March 15th,2021
+    // TODO: This code works as long as existing opinions are in the stage’s local LayerStack.
+    // Relying on the "expanded primIndex" would be a better way to detect the opinions across
+    // composition arc(s). Some more details can be found:
+    // https://groups.google.com/g/usd-interest/c/xTxFYQA_bRs/m/lX_WqNLoBAAJ
+    PXR_NS::SdfLayerHandle strongLayer;
+    for (const auto& spec : propertyStack) {
+
+        // skip if the edit target layer is stronger than the propSpec layer
+        const auto propSpecLayerIndex = findLayerIndex(attr, spec->GetLayer());
+        if (targetLayerIndex <= propSpecLayerIndex) {
+            continue;
+        }
+
+        if (spec) {
+            strongLayer = spec->GetLayer();
+            break;
+        }
+    }
+
+    if (strongLayer) {
+        return false;
+    }
+
+    return true;
```

Comment:
I duplicated the isAttributeEditAllowed() routine here to make it work with the python binding. 
I realize it is a bit wacky but having a python binding for "Ufe/Private/Utils" turned out to be more work. We can chat offline if you think there is a better way to go about this.

Reply:
Yeah, the code duplication is unfortunate.  Would it help if you moved it out of private/Utils and put it somewhere else?  Because now, your Python code isn't testing your C++ code...

Reply:
@ppt-adsk I got rid of the wacky duplicate by moving out the code one level up into "Ufe/Utils" which would work better.

Addressed in : fe8131c1c837ebf7bf5405e7c594df7b5b7b1db4